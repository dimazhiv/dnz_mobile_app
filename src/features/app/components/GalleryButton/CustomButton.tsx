import React from 'react';
import { StyleSheet, Text, TouchableOpacity } from 'react-native';

export type CustomButtonProps = {
  title: string;
  onPress: () => void;
};

function CustomButton({ onPress, title }: CustomButtonProps) {
  return (
    <TouchableOpacity style={styles.button} onPress={onPress}>
      <Text style={styles.buttonText}>{title}</Text>
    </TouchableOpacity>
  );
}



const styles = StyleSheet.create({
  button: {
    backgroundColor: 'blue',
    alignItems: 'center',
    justifyContent: 'center',
    width: 150,
    height: 50,
    borderWidth: 1,
    borderColor: 'blue',
    borderRadius: 10
  },
  buttonText: {
    fontSize: 20,
    color: 'white'
  }
});

export default CustomButton;
