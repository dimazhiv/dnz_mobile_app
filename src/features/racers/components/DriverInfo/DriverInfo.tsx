import React from 'react';
import { StyleSheet, Text, TouchableOpacity, View } from 'react-native';
import { useOpenLink } from '../../../../hooks/hooks';

export type DriverInfoProps = {
  url: string;
  fullName: string;
  dateOfBirth: string;
  nationality: string;
};

function DriverInfo({ url, fullName, dateOfBirth, nationality }: DriverInfoProps) {
  const handleLinkPress = useOpenLink(url);

  return (
    <View style={styles.container}>
      <Text style={styles.name}>{fullName}</Text>
      <Text style={styles.date}>{dateOfBirth}</Text>
      <Text style={styles.nationality}>{nationality}</Text>
      <TouchableOpacity onPress={handleLinkPress}>
        <Text style={styles.linkStyle} numberOfLines={1}>
          more info
        </Text>
      </TouchableOpacity>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center'
  },
  name: {
    fontSize: 32
  },
  date: {
    fontSize: 24
  },
  nationality: {
    fontSize: 24
  },
  linkStyle: {
    fontSize: 36,
    paddingVertical: 50,
    color: 'blue',
    textDecorationLine: 'underline'
  }
});

export default DriverInfo;
