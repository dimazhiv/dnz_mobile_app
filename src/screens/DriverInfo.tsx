import React from 'react';
import { SafeAreaView, StyleSheet } from 'react-native';
import DriverInfoComponent from '../features/racers/components/DriverInfo/DriverInfoContainer';

function DriverInfo() {
  return (
    <SafeAreaView style={styles.container}>
      <DriverInfoComponent />
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
  container: {
    justifyContent: 'center',
    flex: 1
  }
});

export default DriverInfo;
