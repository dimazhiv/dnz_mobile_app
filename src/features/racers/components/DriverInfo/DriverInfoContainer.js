import DriverInfo from './DriverInfo';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { getActiveDriverInfo } from '../../selectors';
import { onDriverNamePress } from '../../sagaActions';

function mapStateToProps(state) {
  const { url, givenName, familyName, dateOfBirth, nationality } = getActiveDriverInfo(state);

  return {
    url,
    fullName: `${givenName} ${familyName}`,
    dateOfBirth,
    nationality
  };
}

function mapDispatchToProps(dispatch) {
  return bindActionCreators(
    {
      onPress: onDriverNamePress
    },
    dispatch
  );
}

export default connect(mapStateToProps, mapDispatchToProps)(DriverInfo);
