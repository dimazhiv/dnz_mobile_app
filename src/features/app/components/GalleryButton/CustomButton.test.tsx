import React from 'react';
import { shallow } from 'enzyme';
import DriverInfo from './DriverInfo';
import CustomButton from "../../../app/components/GalleryButton/CustomButton";

describe('Snapshot testing DriverInfo component', () => {
  test('correct component view', () => {
    const tree = shallow(
      <CustomButton
       title={'Gallery'}
       onPress={()=>{}}/>
    );
    expect(tree).toMatchSnapshot();
  });
});
