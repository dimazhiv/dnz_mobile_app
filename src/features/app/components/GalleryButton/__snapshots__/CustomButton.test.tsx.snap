// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot testing DriverInfo component correct component view 1`] = `
<TouchableOpacity
  onPress={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "red",
      "borderColor": "blue",
      "borderRadius": 10,
      "borderWidth": 1,
      "height": 50,
      "justifyContent": "center",
      "width": 150,
    }
  }
>
  <Text
    style={
      {
        "color": "white",
        "fontSize": 20,
      }
    }
  >
    Gallery
  </Text>
</TouchableOpacity>
`;
